templates/core/terraform/processor_function/function.tf

```diff
@@ -13,24 +19,34 @@ resource "azurerm_function_app" "procesorfunction" {
     https_only                            = true
     FUNCTIONS_WORKER_RUNTIME              = "python"
     FUNCTION_APP_EDIT_MODE                = "readonly"
+    FUNCTIONS_EXTENSION_VERSION           = "3"
     RESOURCE_GROUP_NAME                   = var.resource_group_name
     APP_INSIGHTS_INSTRUMENTATION_KEY      = var.app_insights_instrumentation_key
     VNET_NAME                             = var.core_vnet
     ACI_SUBNET                            = var.aci_subnet
-    CNAB_AZURE_STATE_STORAGE_ACCOUNT_NAME = var.storage_account_name
-    CNAB_AZURE_STATE_STORAGE_ACCOUNT_KEY  = var.storage_account_access_key
-    CNAB_AZURE_STATE_PATH                 = var.storage_state_path
-    CNAB_AZURE_STATE_FILESHARE            = var.storage_state_path
+    CNAB_AZURE_STATE_STORAGE_ACCOUNT_NAME = var.mgmt_storage_account_name
+    SEC_CNAB_AZURE_STATE_STORAGE_ACCOUNT_KEY  = data.azurerm_storage_account.state_storage.primary_access_key
```

Comment:
Do we run into the same issue with lack of support for MSI here?

Reply:
Yep, we need to choose either VNET or MSI with ACI for now. We are doing SPN across the board for simplicity, and then will switch to full MSI once it's supported in VNET